marklogic-data-hub-central/ui/src/pages/EntityTypes.tsx

```diff
@@ -43,18 +47,92 @@ const EntityTypes: React.FC = () => {
           }
 
     }
+
+  //GET all the flow artifacts
+  const getFlows = async () => {
+        try {
+            let response = await axios.get('/api/flows');
+            if (response.status === 200) {
+                setFlows(response.data);
+                console.log('GET flows successful', response);
+            } 
+        } catch (error) {
+            let message = error.response.data.message;
+            console.log('Error getting flows', message);
+        } finally {
+        resetSessionTime();
+        }
+    }
+
+  // POST load data step to new flow
+  const addStepToNew = async () => {
+    try {
+      setIsLoading(true);
+      //let response = await axios.post(`/api/artifacts/????/${flowName}`);
+      
+      //if (response.status === 200) {
+        console.log('POST addStepToNew');
+        setIsLoading(false);
+      //} 
+    } catch (error) {
+        let message = error.response.data.message;
+        console.log('Error while adding load data step to new flow.', message);
+        setIsLoading(false);
+        handleError(error);
+    } finally {
+      resetSessionTime();
+    }
+  }
+
+  // POST load data step to existing flow
+  const addStepToFlow = async (loadArtifactName, flowName) => {
+    let stepToAdd = {
+      "name": loadArtifactName,
+      "stepDefinitionName": "default-ingestion",
+      "stepDefinitionType": "INGESTION",
+      options: {
+        "loadData": { 
+          "name": loadArtifactName
+        }
+      }
+    };
+    try {
+      setIsLoading(true);
+      let url = '/api/flows/' + flowName + '/steps';
+      let body = stepToAdd;
+      let response = await axios.post(url, body);
```

Comment:
We should try and refactor api calls under " marklogic-data-hub-central/ui/src/api/" for anything that gets committed to develop. I will create a story for refactoring existing axios get and post requests in develop branch.

Reply:
Refactoring will be done in separate story

Reply:
Refactoring existing code can be part of a separate story. These are new API requests right?

Reply:
Adding loadData, mapping and other artifacts to flow will all be using the same endpoint which will be created in that new refactoring storing .